addons/barfiller/barfiller.lua

```diff
@@ -0,0 +1,116 @@
+--[[    BSD License Disclaimer
+        Copyright © 2015, Morath86
+        All rights reserved.
+
+        Redistribution and use in source and binary forms, with or without
+        modification, are permitted provided that the following conditions are met:
+
+            * Redistributions of source code must retain the above copyright
+              notice, this list of conditions and the following disclaimer.
+            * Redistributions in binary form must reproduce the above copyright
+              notice, this list of conditions and the following disclaimer in the
+              documentation and/or other materials provided with the distribution.
+            * Neither the name of BarFiller nor the
+              names of its contributors may be used to endorse or promote products
+              derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+        ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+        WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        DISCLAIMED. IN NO EVENT SHALL Morath86 BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'BarFiller'
+_addon.author = 'Morath'
+_addon.version = '0.2.0'
+_addon.commands = {'bf','barfiller'}
+_addon.language = 'english'
+
+--Windower Libs
+config = require('config')
+file = require('files')
+packets = require('packets')
+texts = require('texts')
+
+--BarFiller Libs
+require('statics')
+
+-- Generate Settings Files
+-- Thanks to Byrth & SnickySnacks' BattleMod addon
+settings_table = config.load(defaults)
+config.save(settings_table)
+
+ready = false
+chunk_update = false
+
+-- Make sure character is logged in, and loaded before initializing
+windower.register_event('load',function()
+    if windower.ffxi.get_info().logged_in and windower.ffxi.get_player() then
+        initialize() -- Populate character details
+    end
+end)
+
+-- Delay initialize() for 10 seconds to allow game to download chunks
+windower.register_event('login',function()
+    windower.send_command('wait 10;barfiller clear;')
+end)
+
+-- If you're switching characters this will clear the addon from memory
+windower.register_event('logout',function()
+    windower.send_command('lua r barfiller')
```

Comment:
That was a (very poor) cheat on my part, I've changed it to hide all the ```prim``` and ```text``` objects in my next push.